realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1318,6 +1318,16 @@ void BaseRealSenseNode::publishFrame(rs2::frame f, const ros::Time& t,
     ROS_DEBUG("publishFrame(...)");
     auto& image = images[stream];
 
+    if (copy_data_from_frame) {
+        if (stream != DEPTH) {
+            image.data = (uint8_t*)f.get_data();
+        } else {
+            auto depth_data = (uint16_t*)f.get_data();
+            auto& image_data = reinterpret_cast<uint16_t*&>(image.data);
```

Comment:
Won't this break the data?

e.g. [A, B, C, D] will be cast as [AB, CD]? Or is the data already 16 bit?

Reply:
The data is already 16bit. The problem was that the original code was down casting the depth data into 8-bit data.